### DISCUSSION TITLE
Define release process

### DISCUSSION THREAD

**Discussion by orpheuslummis** [sourcenetwork/defradb#254] [General]
Some points to consider:
- code freeze
- beta, release candidate
- bump up version number, update changelog, tag, push to `main` branch
- release maintenance policy, timeline of support for major versions
- detailing upgrade path between major versions (in light of breaking changes)
- ensure external documentation is updated in lock-step, such as on  https://pkg.go.dev
- update parametric license
- release artifacts
    - binaries for supported architectures https://github.com/golang/go/wiki/MinimumRequirements
    - provide checksums and digital signatures
- target a specific calendar-based release cycle
- push out announcement on social/news channels
- `go mod tidy`

The release process to be written down in a public document in source code or in public documentation.

Some references:
- https://go.dev/doc/modules/release-workflow
- https://github.com/golang/go/wiki/Go-Release-Cycle
- https://www.python.org/dev/peps/pep-0602/

**Comment by jsimnz** [comment]
love that you put this together, was swirling around in my head. Having a formal process will keep things in check, and I think you got all the major items suggested

**Comment by orpheuslummis** [comment]
See internal document about this https://hackmd.io/phaHgXW5SrGjRieVRXBNJg?view
